Bug fixes

abhinav2196 opened the pull request.

abhinav2196 committed:
fix: correct organization name and improve CI workflow reliability

- Fix Docker registry organization name from 'blocksize-capital-gmbh' to 'Blocksize-Capital-GmbH'
- Fix package visibility API organization name to match GitHub organization
- Add error handling for package visibility API calls to prevent CI failures
- Update Trivy action from @master to stable @0.24.0 version
- Add category to SARIF upload to prevent conflicts
- Make package visibility step non-blocking if it fails

This resolves GitHub Actions failures related to organization name mismatches
and improves the overall reliability of the CI pipeline.

abhinav2196 committed:
fix: remove poetry.toml that conflicts with CI virtual environment setup

The poetry.toml file with 'in-project = true' was causing CI failures because:
- CI workflow expects venv at: /artifacts/venv/<name>/
- poetry.toml forces venv at: /artifacts/source/.venv/
- This mismatch caused virtual environment verification to fail

Removing poetry.toml restores the working CI configuration from before commit f147ca5.

abhinav2196 committed:
feat: add flexible development environment setup and improve CI security scanning

Development Environment Improvements:
- Add automated dev-setup.sh script for easy local development configuration
- Configure Poetry to use in-project .venv/ without affecting CI pipelines
- Update README with comprehensive development setup instructions
- Add Poetry configuration troubleshooting guide

CI Pipeline Improvements:
- Fix security scan conditional logic with proper parentheses
- Add security-events permission for SARIF uploads
- Add branch-specific security scanning (table format)
- Improve security scan error handling with exit-code: 0
- Make security scan more robust with better conditionals

The local Poetry configuration uses --local flag so it won't be committed to git
and won't interfere with CI pipeline's custom virtual environment setup.

abhinav2196 committed:
add templates and readmes

abhinav2196 committed:
Update CODEOWNERS

abhinav2196 committed:
fix: resolve dual CI runs and improve workflow conditional logic

Issues Fixed:
- Split ci_success job into separate main/branch versions to prevent waiting for skipped jobs
- Add proper parentheses to all conditional expressions for correct operator precedence
- Include security_scan_branch in docker_branch dependencies for complete workflow
- Prevent conflicting job execution paths that caused one pass/one fail scenario

Technical Details:
- ci_success job was waiting for both docker_main AND docker_branch
- On feature branches, docker_main gets skipped but ci_success still waited for it
- Boolean operator precedence caused main jobs to run when they shouldn't
- This resulted in two parallel execution paths with different outcomes

The fix ensures only one execution path runs per branch type and all dependencies are correct.

abhinav2196 committed:
fix: eliminate duplicate CI runs with unified workflow pipeline

Root cause analysis:
- Complex dual pipeline structure (_main vs _branch jobs) created race conditions
- Broad wildcard trigger 'branches: ["*"]' caused potential conflicts
- Missing jobs in branch pipeline created incomplete coverage
- Separate docker jobs with complex conditionals led to inconsistent results

Solution implemented:
- Unified single pipeline that runs consistently on all branches
- Specific branch triggers instead of wildcard patterns
- Dynamic job behavior based on branch context (coverage, security uploads)
- Complete job coverage for all branches
- Simplified dependency chain eliminates race conditions

This ensures exactly one CI run per commit with consistent pass/fail behavior.

abhinav2196 committed:
fix: prevent duplicate CI runs using skip-duplicate-actions

Problem: Both push and pull_request events trigger when pushing to a branch with an open PR, causing duplicate workflow runs.

Solution: Added skip-duplicate-actions@v5 to intelligently detect and skip duplicate runs by:
- Detecting same content/commit across different trigger events
- Skipping concurrent runs for the same branch/PR
- Preferring pull_request events over push events for better PR experience

This ensures only one CI run executes per unique commit/content change.

abhinav2196 committed:
fix: eliminate dual CI triggers by temporarily removing bug-fixes from push events

Problem:
- Push to bug-fixes branch triggers CI via push event
- Open PR #4 (bug-fixes -> main) triggers CI via pull_request event
- Both events fire simultaneously causing duplicate CI runs
- Repository redirect/rename broke skip-duplicate-actions functionality

Solution:
- Temporarily remove 'bug-fixes' from push triggers while PR #4 is open
- CI will only run via pull_request event (single run, better for PR review)
- Will re-add bug-fixes to push triggers after PR is merged/closed

This ensures exactly one CI run per commit for the duration of the PR.

abhinav2196 committed:
CI pipeline update/ refctored

BSC-aujl commented on .github/CODEOWNERS:
```diff
@@ -0,0 +1,34 @@
+
+# Global rule: All changes need review from the main maintainer
+# Replace "your-github-username" with your actual GitHub username
+* @abhinav2196
```
If you really want to add this file, happy to be added as well

BSC-aujl commented on SECURITY.md:
```diff
@@ -0,0 +1,166 @@
+# Security Policy
+
```
too verbose?

abhinav2196 replied:
Can condense after discussion later in next iterations

abhinav2196 committed:
refactor: improve CI architecture and add BSC-aujl as code owner

Major improvements:

🏗️ CODEOWNERS Update:
- Added @BSC-aujl as additional code owner alongside @abhinav2196
- Enhanced review coverage for all critical files (CI/CD, dependencies, docs)
- Shared responsibility for production-critical components

⚡ CI Architecture Simplification:
- Simplified main workflow orchestrator (main.yml)
- Removed complex conditional logic and ci_success job
- Clean branch-based workflow selection
- Better maintainability and readability

Benefits:
✅ Dual code owner coverage for better code quality
✅ Cleaner, more maintainable CI workflow structure
✅ Reduced complexity in main orchestrator
✅ Enhanced team collaboration and code review process
